sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/codegen/CodeGenerator.scala

```diff
@@ -993,6 +994,15 @@ class CodegenContext {
     }
   }
 
+  /**
+   * Returns the code for subexpression elimination after splitting it if necessary.
+   */
+  def subexprFunctionsCode: String = {
+    // Wholestage codegen does not allow subexpression elimination: in that case, subexprFunctions
```

Comment:
I think this is not true, see `subexpressionEliminationForWholeStageCodegen`. @mgaido91 can you doule check to see if this fix works for whole-stage-codegen as well?

Reply:
Seems like whole-stage-codegen has the same issue, but unfortunately `splitExpressionsWithCurrentInputs` is not completed yet.

I think the right description here is: whole-stage-codegen supports subexpression elimination, but we are not able to split the code for it yet.

Reply:
cc @maropu @viirya to double-check.

Reply:
@cloud-fan yes, sorry, this is not very well explained. I mean the wholestage codegen does not honor `spark.sql.subexpressionElimination.enabled` and even though it is true, it doesn't use `subexprFunctions`. So in this method we don't have to deal with wholestage codegen. Do you have suggestion on rewording this comment?

Reply:
how about `whole-stage-codegen supports subexpression elimination, and is handled by another code path`.

Reply:
thanks

Reply:
currently looks better.

Reply:
looks ok, too.